middleware/authz.js

```diff
@@ -0,0 +1,22 @@
+require("express-async-errors");
+const jwt = require("jsonwebtoken");
+
+module.exports =  (listOfRolesAllowed) => {
+  return async (req, res, next) => {
+    const token = req.header("x-auth-token");
+    if (!token)
```

Comment:
To enhance code modularity, consider creating a separate middleware for token validation. This approach avoids repetitive code and centralizes token handling, making it easier to maintain and reuse across different middleware functions.

Reply:
done